### DISCUSSION TITLE
`Number::parse()` proposal

### DISCUSSION THREAD

**Discussion by Giovani-Freitag** [laravel/framework#49974] [Ideas]
Are there plans to have a parsing helper in the `Number` class?

My app is located in `pt_BR`, and I usually need to use the `NumberFormatter` library to parse inputs like `"10,50"` to `10.5` before performing calculations. When I saw this new helper I thought it could help me in this task.

Currently, I have to do something like this in a lot of places:
```php
$num = (new NumberFormatter(config('app.locale'), NumberFormatter::DECIMAL))
    ->parse($request->input('number'));
```

Using a helper, I could have something just like:
```php
$num = Number::parse($request->input('number'));
```

If it's necessary to work with other formats, it would be possible with named parameters:
```php
$num = Number::parse($request->input('number'), type: NumberFormatter::TYPE_INT32, locale: config('app.locale'));
```

If accessing constants from the `NumberFormatter` library is a problem, maybe there could be specific parses like:
```php
$num = Number::parse($request->input('number'));
$num = Number::parseInt($request->input('number'));
$num = Number::parseFloat($request->input('number'));
```

It's my first post here, I hope it's on the right place 😅

**Comment by codespearhead** [comment]
Do you have the bandwidth to submit a PR with the suggested method?

**Comment by informagenie** [comment]
Hi, 

I have just submitted a [PR ](https://github.com/laravel/framework/pull/55725) along these lines.
